src/structure.js

```diff
@@ -81,6 +82,20 @@ function Structure (options) {
 inherits(Structure, EventEmitter);
 module.exports = Structure;
 
+Structure.prototype._subscribe = function(path, fn) {
```

Comment:
Inside a reference we could do something like this to ease the use (if you feel passing `this._referenceListeners` is  "unclean"/noisy):

```js
var localSubscribe = partial(subscribe, this._referenceListeners, path);

// somewhere else (inside `reference.subscribe`)
this._referenceListeners = localSubscribe(fn);
```